pkg/api/v1/validation/validation.go

```diff
@@ -60,6 +60,15 @@ func ValidateResourceRequirements(requirements *v1.ResourceRequirements, fldPath
 		allErrs = append(allErrs, validateContainerResourceName(string(resourceName), fldPath)...)
 		// Validate resource quantity.
 		allErrs = append(allErrs, ValidateResourceQuantityValue(string(resourceName), quantity, fldPath)...)
+
```

Comment:
I feel the check under `limits` validation can be merged into this logic.
1. If limits are specified and requests aren't specified, requests will equal limits. The logic here will be hit.
2. If limits aren't specified and request is, then this logic will be hit as well.
3. If neither is specified, we do not care.